libmscore/layout.cpp

```diff
@@ -1693,14 +1693,17 @@ System* Score::getNextSystem(LayoutContext& lc)
 void Score::createMMRest(Measure* m, Measure* lm, const Fraction& len)
       {
       int n = 1;
-      for (Measure* mm = m->nextMeasure(); mm; mm = mm->nextMeasure()) {
-            ++n;
-            mm->setMMRestCount(-1);
-            if (mm->mmRest())
-                  undo(new ChangeMMRest(mm, 0));
-            if (mm == lm)
-                  break;
+      if (m != lm) {
```

Comment:
Comparing pointers to determine equality is traditionally not the best way since it doesn't work with copies of the same measure object. It is better to introduce isEqual() method which compares specific measure properties, but unfortunately current architecture doesn't have such properties.

Reply:
@wschweer what do you think whether it is possible to introduce specific ids for score entities? And whether it is necessary or not?

Reply:
I don't see the point. We compare measure pointers all over the place.

Reply:
Yes, we do it and it works. It works until someone create copy of the Measure pointer to keep it somewhere for somewhat reason. After that, two different pointers will exist and point to the same "real" measure, so comparing these two measures which are the same in terms of score, but different in terms of pointers will lead to incorrect comparison result.
Btw, It is not the case right now.